datadog_lambda/dsm.py

```diff
@@ -0,0 +1,41 @@
+from datadog_lambda import logger
+from datadog_lambda.trigger import EventTypes
+
+
+def set_dsm_context(event, event_source):
+
+    if event_source.equals(EventTypes.SQS):
+        _dsm_set_sqs_context(event)
+
+
+def _dsm_set_sqs_context(event):
+    from datadog_lambda.wrapper import format_err_with_traceback
+
+    from ddtrace.internal.datastreams.processor import (
+        DataStreamsProcessor,
+        DsmPathwayCodec,
+    )
+    from ddtrace.internal.datastreams.botocore import (
+        get_datastreams_context,
+        calculate_sqs_payload_size,
+    )
+
+    records = event.get("Records")
+    if records is None:
+        return
+    processor = DataStreamsProcessor()
```

Comment:
According to my undersatnding of ddtrace (so please correct me if I'm wrong), initializing a new `DataStreamsProcessor` object starts a bunch of threads and has a whole ton of initialization code. We want to avoid doing that over and over again. Doing it like

```
from ddtrace.internal.datastreams import data_streams_processor

processor = data_streams_processor()
```

on the other hand ensures that we use the same global processor throughout each invocation.